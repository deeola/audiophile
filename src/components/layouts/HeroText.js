import React from 'react'

function HeroText(props) {
    return (
        <div className='heroText' >
            <p className='productPara'>New Product</p>
            <h1 className='heroMainText'>XX99 Mark II Headphones</h1>
            <p className='productDescr'>Experience natural, lifelike audio and exceptional build quality made for the passionate music enthusiast.</p>
            <button>See product</button>
            
        </div>
    )
}


export default HeroText

