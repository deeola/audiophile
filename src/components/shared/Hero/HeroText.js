import React from 'react'
import { Button } from '../../Buttons/Button'
import { Header, HeroTextControl, NewProductText, SubHeading } from './Herostyling'

const HeroText = ({datatestid}) => {
  return (
    <HeroTextControl data-testid='hero'>
        <NewProductText>NEW PRODUCT</NewProductText>
        <Header data-testid='heroheader'>XX99 MARK II HEADPHONES</Header>
        <SubHeading data-testid='subheader'>Experience natural, lifelike audio and exceptional build quality made for the passionate music enthusiast.</SubHeading>
        <Button datatestid={datatestid} label={'SEE PRODUCT'} link='/xx99-mark-two' />
    </HeroTextControl>
  )
}

export default HeroText

