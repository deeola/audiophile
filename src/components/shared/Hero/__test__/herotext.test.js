import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup, screen, getByText } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import HeroText from "../HeroText";
import { BrowserRouter } from "react-router-dom";
import reactDom from "react-dom";

const MockHeroText = () => {
  return (
    <BrowserRouter>
      <HeroText />
    </BrowserRouter>
  );
};
it("renders heroText correctly", () => {
  const header = document.createElement("header");
  reactDom.render(<MockHeroText />, header);
});

describe("checks tests", () => {
  it("renders heading text correctly", () => {
    const { getByTestId } = render(<MockHeroText />);
    expect(getByTestId("heroheader")).toHaveTextContent(
      'XX99 MARK II HEADPHONES'
    );
  });

  // checks for subheading text

  it("renders subheading text correctly", () => {
    const { getByTestId } = render(<MockHeroText />);
    expect(getByTestId("subheader")).toHaveTextContent(
      'Experience natural, lifelike audio and exceptional build quality made for the passionate music enthusiast.'
    );
  });


});


// create snapshot

it('creates snapshot for hero text section', () => {
    const tree = renderer.create(<MockHeroText />).toJSON()
    expect(tree).toMatchSnapshot();
})
