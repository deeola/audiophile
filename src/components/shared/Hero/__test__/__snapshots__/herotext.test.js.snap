// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates snapshot for hero text section 1`] = `
<div
  className="sc-papXJ ewFfCL"
  data-testid="hero"
>
  <p
    className="sc-ftvSup hvZees"
  >
    NEW PRODUCT
  </p>
  <h1
    className="sc-jqUVSM hFGwfU"
    data-testid="heroheader"
  >
    XX99 MARK II HEADPHONES
  </h1>
  <p
    className="sc-kDDrLX byaiTP"
    data-testid="subheader"
  >
    Experience natural, lifelike audio and exceptional build quality made for the passionate music enthusiast.
  </p>
  <a
    className="sc-bczRLJ lnGMsn"
    color="primary"
    hover="primary"
    href="/xx99-mark-two"
    onClick={[Function]}
  >
    SEE PRODUCT
  </a>
</div>
`;
